# Copyright 2025 Dell Inc. or its subsidiaries. All Rights Reserved.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
---

# Usage: deploy_loki_container.yml
omnia_nfs_share: "{{ oim_shared_path }}/omnia"
dir_permissions_755: "0755"
loki_container_name: "loki"
loki_image: docker.io/grafana/loki
# Tag is fixed for grafana/loki image as of 12-06-2025
loki_image_tag: "3.5.1"
loki_promtail_config_dir: "{{ omnia_nfs_share }}/telemetry/loki_promtail"
container_config_mnt: "/mnt/config"
loki_podman_volumes:
  - { name: "loki_promtail_config", path: "{{ loki_promtail_config_dir }}" }
loki_volumes:
  - "loki_promtail_config:{{ container_config_mnt }}{{ selinux_option }}"
loki_config_file_path:
  - { src: "files/loki-config.yaml", dest: "{{ loki_promtail_config_dir }}", mode: "0644" }
loki_container_success_msg: "The Loki container '{{ loki_container_name }}' container has been successfully deployed."
loki_container_failure_msg: |
  Deployment of the {{ loki_container_name }} container has failed.

  If the container was created but is not in a running state, check the logs using:
    podman logs {{ loki_container_name }}

  Then, run the utility playbook to clean up existing OIM resources:
    ansible-playbook utility/oim_cleanup.yml

  After cleanup, re-run the prepare_oim.yml playbook to attempt the deployment again.

loki_image_pull_fail_msg: |
  The pull of the Loki image '{{ loki_image }}:{{ loki_image_tag }}' has failed.
  To resolve this, please check the following error: {{ image_pull_result.msg }}.
  After addressing the issue, you can re-run the playbook to successfully pull the Loki image.
loki_image_not_found_msg: |
  The Loki image '{{ loki_image }}:{{ loki_image_tag }}' was not found after pull attempt.
  Please check if the image is available and re-run the playbook.
max_retries: 10
delay_count: 5

# Usage: deploy_promtail_container.yml
promtail_container_name: "promtail"
promtail_image: docker.io/grafana/promtail
# Tag is fixed for grafana/promtail image as of 12-06-2025
promtail_image_tag: "3.5.1"
promtail_volumes:
  - "{{ omnia_nfs_share }}:/opt/omnia{{ selinux_option }}"
  - "loki_promtail_config:{{ container_config_mnt }}{{ selinux_option }}"
promtail_config_file_path:
  - { src: "files/promtail-config.yaml", dest: "{{ loki_promtail_config_dir }}", mode: "0644" }
promtail_container_success_msg: "The Promtail container '{{ promtail_container_name }}' container has been successfully deployed."
promtail_container_failure_msg: |
  Deployment of the {{ promtail_container_name }} container has failed.

  If the container was created but is not in a running state, check the logs using:
    podman logs {{ promtail_container_name }}

  Then, run the utility playbook to clean up existing OIM resources:
    ansible-playbook utility/oim_cleanup.yml

  After cleanup, re-run the prepare_oim.yml playbook to attempt the deployment again.

promtail_image_pull_fail_msg: |
  The pull of the Promtail image '{{ promtail_image }}:{{ promtail_image_tag }}' has failed.
  To resolve this, please check the following error: {{ image_pull_result.msg }}.
  After addressing the issue, you can re-run the playbook to successfully pull the Promtail image.
promtail_image_not_found_msg: |
  The Promtail image '{{ promtail_image }}:{{ promtail_image_tag }}' was not found after pull attempt.
  Please check if the image is available and re-run the playbook.
